cmd/seid/cmd/iavl_parser.go

```diff
@@ -37,19 +49,87 @@ func BankParser(key []byte) ([]string, error) {
 		keyItems = append(keyItems, "Balances")
 		// remaining is length prefixed addr + denom
 		remaining := bytes.TrimPrefix(key, banktypes.BalancesPrefix)
-		lengthPrefix, remaining := int(remaining[0]), remaining[1:]
-		keyItems = append(keyItems, fmt.Sprintf("AddrLength: %d", lengthPrefix))
-		accountAddr := remaining[0:lengthPrefix]
-		denom := remaining[lengthPrefix:]
-		bech32Addr, err := sdk.Bech32ifyAddressBytes("sei", accountAddr)
+		items, denom, err := parseLengthPrefixedAddress(remaining)
 		if err != nil {
 			return keyItems, err
 		}
-		keyItems = append(keyItems, fmt.Sprintf("AddrBech32: %s", bech32Addr))
+		keyItems = append(keyItems, items...)
 		keyItems = append(keyItems, fmt.Sprintf("Denom: %s", string(denom)))
 	default:
 		keyItems = append(keyItems, "Unrecognized prefix")
 	}
 
 	return keyItems, nil
 }
+
+func DexParser(key []byte) ([]string, error) {
+	keyItems := []string{}
+	if matched, items, _, err := MatchAndExtractDexAddressPrefixKeys(key); matched {
+		if err != nil {
+			return keyItems, err
+		}
+		keyItems = append(keyItems, items...)
+		return keyItems, nil
+	}
+	switch {
+	case bytes.HasPrefix(key, []byte(dexkeeper.EpochKey)):
+		// do nothing since the key is a string and no other data to be parsed
+	default:
+		keyItems = append(keyItems, "Unrecognized prefix")
+	}
+	return keyItems, nil
+}
+
+func MatchAndExtractDexAddressPrefixKeys(key []byte) (bool, []string, []byte, error) {
+	keyItems := []string{}
+	keysToMatch := []string{
+		// Source of truth: github.com/sei-protocol/sei-chain/x/dex/types/keys.go - contains key constants represented here
+		dextypes.LongBookKey,
+		dextypes.ShortBookKey,
+		dextypes.TriggerBookKey,
+		dextypes.PriceKey,
+		dextypes.TwapKey,
+		dextypes.SettlementEntryKey,
+		dextypes.RegisteredPairKey,
+		dextypes.OrderKey,
+		dextypes.CancelKey,
+		dextypes.AccountActiveOrdersKey,
+		dextypes.NextOrderIDKey,
+		dextypes.NextSettlementIDKey,
+		dextypes.MatchResultKey,
+		dextypes.MemOrderKey,
+		dextypes.MemCancelKey,
+		dextypes.MemDepositKey,
+		dexkeeper.ContractPrefixKey,
+	}
+
+	for _, prefix := range keysToMatch {
+		if bytes.HasPrefix(key, dextypes.KeyPrefix(prefix)) {
+			keyItems = append(keyItems, prefix)
+			remaining := bytes.TrimPrefix(key, dextypes.KeyPrefix(prefix))
+			items, remaining, err := parseLengthPrefixedAddress(remaining)
+			if err != nil {
+				return true, keyItems, remaining, err
+			}
+			keyItems = append(keyItems, items...)
+			if len(remaining) > 0 {
+				keyItems = append(keyItems, fmt.Sprintf("RemainingString: %s", string(remaining)))
+			}
+			return true, keyItems, remaining, nil
+		}
+	}
+	return false, keyItems, key, nil
+}
+
+func parseLengthPrefixedAddress(remainingKey []byte) ([]string, []byte, error) {
+	keyItems := []string{}
+	lengthPrefix, remaining := int(remainingKey[0]), remainingKey[1:]
+	accountAddr := remaining[0:lengthPrefix]
+	remaining = remaining[lengthPrefix:]
+	bech32Addr, err := sdk.Bech32ifyAddressBytes("sei", accountAddr)
```

Comment:
can we reference this https://github.com/sei-protocol/sei-chain/blob/master/app/params/config.go#L21 instead?

Reply:
made this and the above fix in https://github.com/sei-protocol/sei-chain/pull/562, can we review that instead since it also contains the staking parser logic